[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda

etaf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #173779
* #161940
* #161939
* #161938
* #160729
* #160706
* #160688
* #160687
* #160686
* __->__ #160685

This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.lin@intel.com>



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @chenyang78

etaf committed:
[Inductor XPU GEMM] Step 3/N: Move cutlass files from torch/_inductor/codegen/cuda to
torch/_inductor/codegen/cutlass.

Signed-off-by: xinan.lin <xinan.lin@intel.com>

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 3/N: Move cutlass files from torch/_inductor/codegen/cuda to"


torch/_inductor/codegen/cutlass.

Signed-off-by: xinan.lin <xinan.linintel.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 3/N: Move cutlass files from torch/_inductor/codegen/cuda to"


torch/_inductor/codegen/cutlass.

Signed-off-by: xinan.lin <xinan.linintel.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 3/N: Move cutlass files from torch/_inductor/codegen/cuda to"


torch/_inductor/codegen/cutlass.

Signed-off-by: xinan.lin <xinan.linintel.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 3/N: Move cutlass files from torch/_inductor/codegen/cuda to"


torch/_inductor/codegen/cutlass.

Signed-off-by: xinan.lin <xinan.linintel.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 3/N: Move cutlass files from torch/_inductor/codegen/cuda to"


torch/_inductor/codegen/cutlass.

Signed-off-by: xinan.lin <xinan.linintel.com>



[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

henrylhtsang commented:
What is the motivation to do so? While I agree the name change makes sense, it is pretty disruptive

henrylhtsang commented:
> In our design，The Intel path is one of the device backend of Inductor CUTLASS，and should always align the user API with nv-cutlass. So there won't be any dependency on nv-cutlass.

ok let me be extra careful and just noob question: say you are templating with some cutlass code. If nvidia changes anything in cutlass, will this intel cutlass backend be affected?

etaf commented:
> > In our design，The Intel path is one of the device backend of Inductor CUTLASS，and should always align the user API with nv-cutlass. So there won't be any dependency on nv-cutlass.
> 
> 
> 
> ok let me be extra careful and just noob question: say you are templating with some cutlass code. If nvidia changes anything in cutlass, will this intel cutlass backend be affected?

Hi， @henrylhtsang ：
We maintain an Intel Cutlass repository which provides the Cutlass header files. This repository is continuously synced with the NVIDIA Cutlass upstream to maintain the same user interface.

Regarding your concern about changes in the NVIDIA repository: the Intel Cutlass repository will not be affected immediately because we use the Intel Cutlass repo. However, if there is a public API change in the NVIDIA repo that impacts the Inductor's Cutlass template, we will first implement a workaround in Inductor and then sync the Intel Cutlass repository. Such situations are rare, however, since our Intel Cutlass repo is frequently synced with the NVIDIA upstream.

Our update process is similar to how Intel Triton syncs with OpenAI's Triton.

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 2/N: Move out cutlass files from torch/_inductor/codegen/cuda"


This PR is the second step toward implementing RFC https://github.com/pytorch/pytorch/issues/160175.
Currently, all Cutlass-related code are located in `torch/_inductor/codege/cuda` This PR moves those file to `torch/_inductor/codegen/cutlass` so they can be shared and reused by XPU as well.

Signed-off-by: xinan.lin <xinan.linintel.com>



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf commented:
@pytorchbot drci

henrylhtsang reviewed (approved):
approve to get meta signal
